src/unasync/__init__.py

```diff
@@ -98,6 +119,7 @@ def unasync_file(filepath, fromdir, todir):
         tokens = tokenize(f)
         tokens = unasync_tokens(tokens)
         result = untokenize(tokens)
+        result = substitute(result, substitutions)
```

Comment:
Hmm, I would have expected we'd run the substitutions while we're working at the token-level, not after we go back down to raw text, similar to how we handle `__anext__` → `__next__`. What do you think?

Reply:
Thanks for the review!

So that works for say `AsyncIterator` -> `Iterator`, but how do you transform `with AsyncPoolManager(backend="trio") as http:` into `with PoolManager() as http:` at the token level?

I guess we could use `with PoolManager(backend="sync")` and then the token-level transformation would suffice, but I expect to see more complicated things to change in the future, such as usage of pytest.mark.parametrize in async tests (to test again multiple backends) that we would want to remove in sync tests.

But maybe I'm optimize prematurely at this point? We can wait to see how the work of @agronholm is going to change our tests to take a decision here.

Reply:
I was expecting we'd be transforming `with AsyncPoolManager(backend=backend)` into `with PoolManager(backend=backend)`, where `backend` is a value coming from pytest's parametrization magic.

Reply:
Or in lots of cases, we can probably leave out the backend argument entirely, because sniffio will fill it in appropriately?

Reply:
But then how does it know whether to use anyio with its own trio backend, or the native trio backend?

Reply:
@agronholm Those are identical from the user's PoV, aren't they? Is there a reason we'd want to expose that option in the public API?

Reply:
Not really identical since anyio uses its own networking code, independent from trio. And, if it were identical, we would not have any use for a separate trio backend in urllib3, would we?

Reply:
I guess it would help here if sniffio detected anyio/multio separately from trio/asyncio/curio?

@agronholm Can you share the way async tests will look like after your current work on testing multiple backends?

Reply:
But identical in the sense that there's no reason that users should choose one in some situations and the other in other situations, right?

> we would not have any use for a separate trio backend in urllib3, would we?

Yeah, I don't think it's clear yet exactly how the backends should shake out. From the user's point of view though, I don't think they care whether we use anyio or not, right? That should be totally invisible, as long as we somehow support their async loop somehow.

Reply:
What you're saying is that right now we should fall back on anyio for backends we don't support (curio and asyncio), but still use our trio backend as long as it exists? That makes sense to me

Reply:
> But identical in the sense that there's no reason that users should choose one in some situations and the other in other situations, right?

I haven't completely thought this through, but I guess that would work.